### ISSUE TITLE
K8SSAND-1501 ⁃ cass-operator ConfigMap image_config.yaml is broken

### ISSUE THREAD

**Issue by Gellardo** [k8ssandra/k8ssandra#1404]
## Bug Report

**Describe the bug**
Cass-operator configmap uses `imagePullSecrets` ([here](https://github.com/k8ssandra/k8ssandra/blob/main/charts/cass-operator/templates/configmap.yaml#L36)) even though cass-operator config is named `imagePullSecret` ([here](https://github.com/k8ssandra/cass-operator/blob/master/apis/config/v1beta1/imageconfig_types.go#L40)) and in fact can't handle more than one secret name.

Also `imageRegistry` ([here](https://github.com/k8ssandra/k8ssandra/blob/main/charts/cass-operator/templates/configmap.yaml#L33)) and `imagePullPolicy` are indented too much.

**To Reproduce**

1. Go to `charts/cass-operator`
1. `helm template . --set 'image.registryOverride=registry,imagePullSecrets[0].name=pull-secret'` > output.yaml
1. Find the `ConfigMap`

```
kind: ConfigMap
metadata:
  name: release-name-cass-operator-manager-config
  [...]
apiVersion: v1
data:
  controller_manager_config.yaml: |
    [...]
  image_config.yaml: |
    apiVersion: config.k8ssandra.io/v1beta1
    kind: ImageConfig
    images:
      system-logger: "k8ssandra/system-logger:v1.10.0"
      config-builder: "datastax/cass-config-builder:1.0.4-ubi7"
      imageRegistry: asdf
    imagePullSecrets:
        - name: hi
      imagePullPolicy: IfNotPresent
    defaults:
      [...]
```

**Expected behavior**
It is possible to add multiple imagePullSecrets to the pods generated by the operator. This is probably not possible without changes to the operator itself.

Pure Chart fix: expected output from my understanding of the current cass-operator config:

```diff
kind: ConfigMap
metadata:
  name: release-name-cass-operator-manager-config
  [...]
apiVersion: v1
data:
  controller_manager_config.yaml: |
    [...]
  image_config.yaml: |
    apiVersion: config.k8ssandra.io/v1beta1
    kind: ImageConfig
    images:
      system-logger: "k8ssandra/system-logger:v1.10.0"
      config-builder: "datastax/cass-config-builder:1.0.4-ubi7"
-     imageRegistry: asdf
+   imageRegistry: asdf
-   imagePullSecrets:
-       - name: hi
+   imagePullSecret:
+       name: hi
-     imagePullPolicy: IfNotPresent
+   imagePullPolicy: IfNotPresent
    defaults:
      [...]
```

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-1501) by [Unito](https://www.unito.io)
┆friendlyId: K8SSAND-1501
┆priority: Medium

**Comment by jsanda** [comment]
Hi @Gellardo 

Thanks for opening the issue.

>It is possible to add multiple imagePullSecrets to the pods generated by the operator. This is probably not possible without changes to the operator itself.

That's right. It will require a change in cass-operator. I went ahead and created https://github.com/k8ssandra/cass-operator/issues/341.

This may be fixed by #1421.

**Comment by jsanda** [comment]
This was fixed by https://github.com/k8ssandra/k8ssandra/pull/1421.
